### ISSUE TITLE
Obtaining Identity of user in DocumentFilter

### ISSUE THREAD

**Issue by arrkaye** [domaindrivendev/Swashbuckle#1104]
Hi,

I've written a DocumentFilter to filter out docs based on whether a user has certain claims.

I can't find any way to get the actual user principal.

Other threads have used 'Thread.CurrentPrincipal' but I can't see where this is. Has this been changed in a newer version?

Thanks.

**Comment by evenflow58** [comment]
Add this to your Startup -> ConfigureServices
`services.AddScoped<IHttpContextAccessor, HttpContextAccessor>();`
You can now get http and auth like so
```
var http = this._provider.GetRequiredService<IHttpContextAccessor>();
var auth = this._provider.GetRequiredService<IAuthorizationService>();
```

I got this code from [this link](https://github.com/jenyayel/SwaggerSecurityTrimming/blob/master/src/V2/SwaggerAuthorizationFilter.cs) along with some functions to check for authorization.

**Comment by vytautas-pranskunas-** [comment]
Yes, but logged in user and his claims are not visible for some reason during DocumentFilter apply. However they are perfectly visible when hitting endpoints from Swagger afterwards.

Any ideas why?
